cmd/genbindings/emitgo.go

```diff
@@ -716,13 +720,41 @@ import "C"
 
 		if len(e.Entries) > 0 {
 
-			ret.WriteString("const (\n")
+			var smallvalues []string
 
 			for _, ee := range e.Entries {
-				ret.WriteString(titleCase(cabiClassName(goEnumShortName+"::"+ee.EntryName)) + " " + goEnumName + " = " + ee.EntryValue + "\n")
+
+				isBigInt := false
+
+				if e.UnderlyingType.IntType() {
+					// Int-type enums need special handling in case of 64-bit
+					// overflow, that would prevent using miqt on 32-bit platforms
+
+					entryValueI64, err := strconv.ParseInt(ee.EntryValue, 10, 64)
+					if err != nil {
+						panic("Enum " + ee.EntryName + " has non-parseable integer value")
+					}
+
+					if entryValueI64 >= math.MaxInt32 || entryValueI64 <= math.MinInt32 {
```

Comment:
This can technically drop the equality portions.

```suggestion
					if entryValueI64 > math.MaxInt32 || entryValueI64 < math.MinInt32 {
```